pyro/poutine/infer_config_poutine.py

```diff
@@ -0,0 +1,61 @@
+from __future__ import absolute_import, division, print_function
+
+from .poutine import Messenger, Poutine
+
+
+class InferConfigMessenger(Messenger):
+    """
+    Modifies contents of the infer kwarg at sample sites
+    """
+    def __init__(self, config_fn):
+        """
+        :param config_fn: a callable taking a site and returning an infer dict
+
+        Constructor. Doesn't do much, just stores the stochastic function
+        and the config_fn.
+        """
+        super(InferConfigMessenger, self).__init__()
+        self.config_fn = config_fn
+
+    def _pyro_sample(self, msg):
+        """
+        :param msg: current message at a trace site.
+
+        If self.config_fn is not None, calls self.config_fn on msg
+        and stores the result in msg["infer"].
+
+        Otherwise, implements default sampling behavior
+        with no additional effects.
+        """
+        msg["infer"].setdefault("enumerate", {})
```

Comment:
Sorry, I think the `enumerate` just shouldn't be there. I think you can simply delete this line.